packages/owl-bot-post-processor/test/owl-core.ts

```diff
@@ -155,4 +157,81 @@ describe('core', () => {
       assert.strictEqual(build.summary, '1 steps failed 🙁');
     });
   });
+  describe('getOwlBotLock', () => {
+    it('reads .OwlBot.lock.yaml and returns parsed YAML', async () => {
+      const prData = {
+        data: {
+          head: {
+            ref: 'my-feature-branch',
+            repo: {
+              full_name: 'bcoe/example',
+            },
+          },
+        },
+      };
+      const config = `docker:
+  image: node
+  digest: sha256:9205bb385656cd196f5303b03983282c95c2dfab041d275465c525b501574e5c`;
+      const content = {
+        data: {
+          content: Buffer.from(config, 'utf8').toString('base64'),
+          encoding: 'base64',
+        },
+      };
+      const octokit = ({
+        pulls: {
+          get() {
+            return prData;
+          },
+        },
+        repos: {
+          getContent() {
+            return content;
+          },
+        },
+      } as any) as InstanceType<typeof Octokit>;
+      const lock = await core.getOwlBotLock('bcoe/test', 22, octokit);
+      assert.strictEqual(lock.docker.image, 'node');
+      assert.strictEqual(
+        lock.docker.digest,
+        'sha256:9205bb385656cd196f5303b03983282c95c2dfab041d275465c525b501574e5c'
+      );
+    });
+    it('throws error if config is invalid', async () => {
+      const prData = {
+        data: {
+          head: {
+            ref: 'my-feature-branch',
+            repo: {
+              full_name: 'bcoe/example',
+            },
+          },
+        },
+      };
+      const config = `no-docker-key:
+      image: node
+      digest: sha256:9205bb385656cd196f5303b03983282c95c2dfab041d275465c525b501574e5c`;
+      const content = {
+        data: {
+          content: Buffer.from(config, 'utf8').toString('base64'),
+          encoding: 'base64',
+        },
+      };
+      const octokit = ({
+        pulls: {
+          get() {
+            return prData;
+          },
+        },
+        repos: {
+          getContent() {
+            return content;
+          },
+        },
+      } as any) as InstanceType<typeof Octokit>;
+      assert.rejects(async () => {
```

Comment:
Is the async wrapper here needed?  Wouldn't this be the same as `assert.rejects(core.getOwlBotLock('bcoe/test', 22, octokit), /batman/)`?

Reply:
Good catch 👍